### ISSUE TITLE
Filter out bad characters for elastic beanstalk description

### ISSUE THREAD

**Issue by dloiko** [travis-ci/dpl#462]
Ran into this strange issue where our travis builds were failing with the following message:

```
/home/travis/.rvm/gems/ruby-1.9.3-p551/gems/aws-sdk-v1-1.66.0/lib/aws/core/client.rb:375:in `return_or_raise': Invalid unicode xml character in CreateApplicationVersionMessage.Description at index 74 (AWS::ElasticBeanstalk::Errors::InvalidParameterValue)
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/aws-sdk-v1-1.66.0/lib/aws/core/client.rb:476:in `client_request'
    from (eval):3:in `create_application_version'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/provider/elastic_beanstalk.rb:138:in `create_app_version'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/provider/elastic_beanstalk.rb:45:in `push_app'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/provider.rb:143:in `block in deploy'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/cli.rb:41:in `fold'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/provider.rb:143:in `deploy'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/cli.rb:32:in `run'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/lib/dpl/cli.rb:7:in `run'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.16/bin/dpl:5:in `<top (required)>'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `load'
    from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `<main>'
```

Turns out the last merge commit had a message with the following character: 👍 
Dpl was attempting to append this commit message to the version description of the elastic beanstalk application, and failing every time. It would be nice if there was some filtering of characters in the application version description to line up with the requirements of elastic beanstalk application version descriptions. Or at least a slightly more clear error message.

**Comment by tzaitsev** [comment]
Same issue here with 🎉. Would be great to get it resolved and allow our developers to properly emote their commits.  🏍 without crashing the beanstalk upload.

**Comment by thisandagain** [comment]
I believe https://github.com/travis-ci/dpl/pull/600 will resolve this issue.

**Comment by BanzaiMan** [comment]
Did some research, found that the #600 is not quite there yet. https://github.com/aws/aws-sdk-ruby/issues/1502 will hopefully clarify which characters should be rejected.

Thank you for your patience.

**Comment by Jolg42** [comment]
Same problem. No solution yet?

**Comment by BanzaiMan** [comment]
I've commented on https://github.com/aws/aws-sdk-ruby/issues/1502 again.

**Comment by Jolg42** [comment]
I think the issue is still not resolved...!

**Comment by jjwatson** [comment]
Having this problem now D:

**Comment by hazartilirot** [comment]
Folks, believe me or not, I'm in 2022 - the same issue! I've already tried to deployed a couple of times each failed with an error:
Invalid unicode xml character in CreateApplicationVersionMessage.Description at index (Aws::ElasticBeanstalk::Errors::InvalidParameterValue)


Thanks for the solution. My commit message has a rocket as a symbol. Once I removed it - the job is done! Ha-ha. No more rockets in a deployment stage.  ¯\_(ツ)_/¯
